TLM/TLM/Manager/Impl/AdvancedParkingManager.cs

```diff
@@ -654,336 +915,583 @@ public class AdvancedParkingManager : AbstractFeatureManager, IAdvancedParkingMa
+                                                                  float sqrDistToParkedVehicle,
+                                                                  bool extendedLogParkingAi,
+                                                                  bool usesPublicTransport,
+                                                                  IExtBuildingManager extBuildingMan) {
+            if (usesCar) {
+                // parked car should be reached now
+                Log._DebugIf(
+                    logParkingAi,
+                    () => $"AdvancedParkingManager.OnCitizenPathFindSuccess({instanceId}): " +
+                    $"Path for citizen instance {instanceId} contains passenger car section and " +
+                    "citizen should stand in front of their car.");
+
+                if (extInstance.atOutsideConnection) {
+                    switch (extInstance.pathMode) {
+                        // car path calculated starting at road outside connection: success
+                        case ExtPathMode.CalculatingCarPathToAltParkPos: {
+                            extInstance.pathMode = ExtPathMode.DrivingToAltParkPos;
+                            extInstance.parkingPathStartPosition = null;
+                            if (logParkingAi) {
+                                Log._Debug(
+                                    $"AdvancedParkingManager.OnCitizenPathFindSuccess({instanceId}): " +
+                                    "Path to an alternative parking position is READY! " +
+                                    $"CurrentPathMode={extInstance.pathMode}");
                             }
-                        } else {
-                            // path does not contain a car section: path can be reused for walking
-#if DEBUG
-                            if (debug)
-                                Log._Debug($"AdvancedParkingManager.OnCitizenPathFindSuccess({instanceId}): A direct car path OR initial path was queried that does not contain a car section. Switching path mode to walking.");
-#endif
 
-                            if (usesPublicTransport) {
-                                // decrease public tranport demand
-                                if (instanceData.m_sourceBuilding != 0) {
-                                    extBuildingMan.RemovePublicTransportDemand(ref extBuildingMan.ExtBuildings[instanceData.m_sourceBuilding], (uint)GlobalConfig.Instance.ParkingAI.PublicTransportDemandUsageDecrement, true);
-                                }
-                                if (instanceData.m_targetBuilding != 0) {
-                                    extBuildingMan.RemovePublicTransportDemand(ref extBuildingMan.ExtBuildings[instanceData.m_targetBuilding], (uint)GlobalConfig.Instance.ParkingAI.PublicTransportDemandUsageDecrement, false);
-                                }
-                                extCitizen.transportMode |= ExtTransportMode.PublicTransport;
```

Comment:
was this removed?

Reply:
I believe Resharper found that `if (usesPublicTransport)` above is checked for `true`, and here it is always `false`, and suggested that code is dead.

Reply:
I double checked, that code is still there just reformatted so that diff shows it as new.